src/lib/AsComponent.coffee

```diff
@@ -0,0 +1,77 @@
+#     NoFlo - Flow-Based Programming for JavaScript
+#     (c) 2018 Flowhub UG
+#     NoFlo may be freely distributed under the MIT license
+getParams = require 'get-function-params'
+{Component} = require './Component'
+
+# ## asComponent generator API
+#
+# asComponent is a helper for turning JavaScript functions into
+# NoFlo components.
+#
+# Each call to this function returns a component instance where
+# the input parameters of the given function are converted into
+# NoFlo inports, and there are `out` and `error` ports for the
+# results of the function execution.
+#
+# Variants supported:
+#
+# * Regular synchronous functions: return value gets sent to `out`. Thrown errors get sent to `error`
+# * Functions returning a Promise: resolved promises get sent to `out`, rejected promises to `error`
+# * Functions taking a Node.js style asynchronous callback: `err` argument to callback gets sent to `error`, result gets sent to `out`
+#
+# Usage example:
+#
+#     exports.getComponent = -> noflo.asComponent Math.random,
+#       description: 'Create a random number'
+exports.asComponent = (func, options) ->
+  hasCallback = false
+  params = getParams(func).filter (p) ->
+    return true unless p.param is 'callback'
+    hasCallback = true
+    false
+
+  c = new Component options
+  for p in params
+    portOptions =
+      required: true
+    unless typeof p.default is 'undefined'
```

Comment:
Hm... This will make a component interface change between ES6->ES5. In ES6 defaults with be presents, input is not required. But when transpiling to ES5 no defaults, and must send inputs in the graph, otherwise network will not start?

Reply:
Tempted to say that we should not support this feature...

Reply:
It is actually a bit worse than that. The defaults will work in ES5, but the arguments with defaults are not visible. So it'll always send the default and you won't get a port.

Reply:
Not a major issue as most browsers support ES6 now.

Reply:
With docs its OK